File: JavaScript/8js 1/feast.js
/*
|--------------------------------------------------------------------------
| The Feast of Many Beasts
|--------------------------------------------------------------------------
|
*/

function feast(beast, dish) {
   return beast[0] == dish[0] && beast.substr(-1) == dish.substr(-1);
}


const feast = (...args) => /^(.).*(.),\1.*\2$/.test(args);
const feast = (a, b) => a[0] + [...a].pop() == b[0] + [...b].pop()
const feast = (b, d) => d.startsWith(b[0]) && d.endsWith(b[b.length - 1])


function feast(beast, dish) {
   return beast[0] === dish[0] && beast.slice(-1) === dish.slice(-1);
}


function feast(beast, dish) {
   return beast[0] === dish[0] && beast[beast.length - 1] === dish[dish.length - 1]
}


console.log(feast("great blue heron", "garlic naan"), true)
console.log(feast("chickadee", "chocolate cake"), true)
console.log(feast("brown bear", "bear claw"), false)

File: Python/8py 1/feast.py
"""
|--------------------------------------------------------------------------
| The Feast of Many Beasts
|--------------------------------------------------------------------------
|
"""

def feast(beast, dish):
   return beast[0] == dish[0] and beast[-1] == dish[-1]


feast = lambda beast, dish: all([beast[i] == dish[i] for i in [0,-1]])


def feast(beast, dish):
   return beast.startswith(dish[0]) and beast.endswith(dish[-1])


def feast(b, d):
   return b[::len(b)-1] == d[::len(d)-1]


print(feast("great blue heron", "garlic naan"), True)
print(feast("chickadee", "chocolate cake"), True)
print(feast("brown bear", "bear claw"), False)